.pylintrc

```diff
@@ -0,0 +1,38 @@
+[FORMAT]
+max-line-length=120
+
+[BASIC]
+const-rgx=(([A-Z_][A-Za-z0-9_]*)|(__.*__))$
+
+[MISCELLANEOUS]
+
+# do not remove this. This line makes pylint to ignore FIXME,XXX,TODO
+notes=
+
+[MESSAGES CONTROL]
+# W1202 - Use % formatting in logging functions and pass the % parameters as arguments
```

Comment:
Please explain why you disabled the following checks:
* W1202 - Use % formatting in logging functions and pass the % parameters as arguments
* C0330 - wrong continued indentation
* W0107 - Unnecessary pass statement
* W0641 - Possibly unused variable
* C0103 - Variable name doesn't conform to snake_case naming style
* W0613 - Unused argument
* W0612 - Unused variable
* C1801 - Do not use `len(SEQUENCE)` to determine if a sequence is empty

Reply:
* W1202 -     * Use % formatting in logging functions and pass the % parameters as arguments
And I think that formatting with `.format` is more convenient. But it's pretty easy to fix, so if you think it should be done, I'll do it.


* C0330 -     * wrong continued indentation 
In places, where I say this warning, pylint advised worse indentation, than current. And also pylint advised indentation differs from indentation advised by pycharm.


* W0107 - Unnecessary pass statement
```
class ExecutionTimeoutException(Exception):
    pass
```
In such cases pylint tells to remove `pass`. I think that it is better to leave it.


* W0641 - Possibly unused variable
Variable was actually used
For example, https://cc.epm.kharlamov.biz/jenkins/job/Build_Legion_Artifacts/545/warnings48Result/source.-4005345589716724829/#32

* C0103 - Variable name doesn't conform to snake_case naming style
This code also disables warning `Constant name doesn't conform to UPPER_CASE naming style`
Pylint says that all variables that are in global scope, should be upper case, which is not always true.
Also, he says that all variables that consist of 1 or 2 letters, doesn't conform to snake_case. But in my opinion, sometimes it is OK to have variable of 1 or 2 letters.
For example `for k, v in dict.items()` if dict is not very big or if it contains some simple keys and values. Or `df` for pandas dataframe, etc.

* W0613 - Unused argument
This probably should be fixed instead of disabling this warning.
But in some places it seemed that this unused argument could be used in future.

* W0612 - Unused variable
Pylint shows this warning in cycles like
`for event, new_data in self.watch():` 
where `new_data` is not used. So fix would be to replace `new_data` with `_`.
But if you want to use such variable in future, you probably would need to spent some time to figure out what in that variable `_`.
I mean that in such places you can`t just get rid of this variable anyway.

* C1801 - Do not use `len(SEQUENCE)` to determine if a sequence is empty
This warning probably better to fix.

Reply:
* W1202, C0330, W0107, C0103 - ok
* W0641 - rewrite using format or f strings.
* W0613 - add pylint comment in such places
* C0103 - discussed in skype